src/include/network/lan/hardware.rb

```diff
@@ -51,19 +51,26 @@ def initialize_network_lan_hardware(include_target)
       Yast.include include_target, "network/lan/cards.rb"
 
       @hardware = nil
+    end
 
-      @widget_descr_hardware = {
-        "HWDIALOG" => {
-          "widget"            => :custom,
-          "custom_widget"     => ReplacePoint(Id(:hw_content), Empty()),
-          "init"              => fun_ref(method(:initHwDialog), "void (string)"),
-          "handle"            => fun_ref(method(:handleHW), "symbol (string, map)"),
-          "store"             => fun_ref(method(:storeHW), "void (string, map)"),
-          "validate_type"     => :function,
-          "validate_function" => fun_ref(method(:validate_hw), "boolean (string, map)"),
-          "help"              => initHelp
-        }
+    def widget_descr_hardware
+      widget_descr = {
+        "widget"        => :custom,
+        "custom_widget" => ReplacePoint(Id(:hw_content), Empty()),
+        "init"          => fun_ref(method(:initHwDialog), "void (string)"),
+        "handle"        => fun_ref(method(:handleHW), "symbol (string, map)"),
+        "store"         => fun_ref(method(:storeHW), "void (string, map)"),
+        "help"          => initHelp
       }
+
+      # validation function currently checks user's input in :ifcfg_name widget
+      # However this widget is present only when adding new device
+      if isNewDevice
```

Comment:
We are modifying the widget in case of a NewDevice not adding a validation, so I don't see an inconsistency in that part.

We could also just return true in `validation_hw` in case of `!isNewDevice` just in case for some reason that method could be called in other scenarios but that method is so binding to the widget that don't know if should be needed.